### PR TITLE
Add arp entries

### PR DESCRIPTION
**Pull request by glazychev-art**
This PR is related to AWS integration

Signed-off-by: Artem Glazychev <artem.glazychev@xored.com>

### CODE REVIEW

**Comment by glazychev-art** [comment]
@edwarnicke 
This PR is related to https://github.com/networkservicemesh/integration-k8s-aws/issues/1
I have tested and this approach fixed the problem with remote cases.

Could you please take a look share you thoughts?

**Comment by denis-tingaikin** [comment]
@glazychev-art Do we have a similar work with arps in the monorepo?

**Comment by glazychev-art** [comment]
Yes, we have. 
I left here the links to the monorepo: https://github.com/networkservicemesh/integration-k8s-aws/issues/1#issuecomment-838517868

File: internal/vppinit/vppinit.go [hunk 1]
```diff
@@ -157,6 +166,53 @@ func createAfPacket(ctx context.Context, vppConn api.Connection, link netlink.Li
...
+		now := time.Now()
+		line, err := reader.ReadString('\n')
+		if err != nil {
+			break
+		}
+		// Skip the first line with headers
+		if l == 0 {
+			continue
+		}
+		parts := strings.Fields(strings.TrimSpace(line))
```
Comment (denis-tingaikin) [inline internal/vppinit/vppinit.go:187]:
```suggestion
		fields := strings.Fields(strings.TrimSpace(line))
```

Also we could declare a consts that can remove magic indexes.

For example:

```go
const (
	ipAddr int = iota
	hwType
	flags
	hwAddr
	mask
	device
)
```

File: internal/vppinit/vppinit.go [hunk 2]
```diff
@@ -157,6 +166,53 @@ func createAfPacket(ctx context.Context, vppConn api.Connection, link netlink.Li
...
+
+		mac, err := net.ParseMAC(strings.TrimSpace(parts[3]))
```
Comment (denis-tingaikin) [inline internal/vppinit/vppinit.go:189]:
```suggestion
		mac, err := net.ParseMAC(strings.TrimSpace(fields[hwAddr]))
```

**Review by denis-tingaikin — APPROVED** [review]

**Comment by edwarnicke** [comment]
Is the issue here that AWS is statically provisioning IP Neighbor entries for the Pod interface and not responding to APR/ND requests on the link?

File: internal/vppinit/vppinit.go [hunk 3]
```diff
@@ -76,6 +80,11 @@ func LinkToAfPacket(ctx context.Context, vppConn api.Connection, tunnelIP net.IP
 		WithField("duration", time.Since(now)).
 		WithField("vppapi", "SwInterfaceSetFlags").Debug("completed")
 
+	err = addArp(ctx, vppConn, swIfIndex)
```
Comment (edwarnicke) [inline internal/vppinit/vppinit.go:83]: Please call this addIPNeighbor - we want to cover more than just arp :)

File: internal/vppinit/vppinit.go [hunk 4]
```diff
@@ -157,6 +166,53 @@ func createAfPacket(ctx context.Context, vppConn api.Connection, link netlink.Li
 	return afPacketCreateRsp.SwIfIndex, nil
 }
 
+func addArp(ctx context.Context, vppConn api.Connection, swIfIndex interface_types.InterfaceIndex) error {
+	f, err := os.OpenFile("/proc/net/arp", os.O_RDONLY, 0600)
```
Comment (edwarnicke) [inline internal/vppinit/vppinit.go:170]: Please use:

https://pkg.go.dev/github.com/vishvananda/netlink#NeighList

To get the list of both IPv4 and IPv6 Neighbors.

Comment (edwarnicke) [reply internal/vppinit/vppinit.go:170]: Also... we should probably take some care to only add ARP entries that are in state [PERMANENT](https://github.com/vishvananda/netlink/blob/v1.1.0/neigh_linux.go#L40).  I presume these AWS neighbor entries are in state PERMANENT?

File: internal/vppinit/vppinit.go (hunk 1 above)
Comment (denis-tingaikin) [reply internal/vppinit/vppinit.go:187]: Lets follow https://github.com/networkservicemesh/cmd-forwarder-vpp/pull/186#discussion_r631096359

File: internal/vppinit/vppinit.go (hunk 2 above)
Comment (denis-tingaikin) [reply internal/vppinit/vppinit.go:189]: Lets follow https://github.com/networkservicemesh/cmd-forwarder-vpp/pull/186#discussion_r631096359

File: internal/vppinit/vppinit.go (hunk 3 above)
Comment (glazychev-art) [reply internal/vppinit/vppinit.go]: Yep, fixed

File: internal/vppinit/vppinit.go (hunk 4 above)
Comment (glazychev-art) [reply internal/vppinit/vppinit.go]: You are right! Fixed

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**